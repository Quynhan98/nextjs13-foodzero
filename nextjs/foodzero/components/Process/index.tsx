import Image from 'next/image'
import { Box, Flex, Heading, Text } from '@chakra-ui/react'

// Constants
import { SOPHISTICATED_PROCESS } from '@constants/index'

const Process = () => {
  return (
    <>
      <Flex
        flexDirection={{ base: 'column', md: 'row' }}
        justifyContent={{ base: 'center', md: 'space-between' }}
        gap={{ base: '10px', md: '102px' }}
        pt={{ base: '30px', md: '174px' }}
        alignItems="center"
        margin={{ base: '0 auto', md: 'unset' }}
      >
        <Box
          width={{ base: '277px', md: '634px', '2xl': '792px' }}
          height={{ base: '233px', md: '447px', '2xl': '638px' }}
          position="relative"
        >
          <Image
            fill
            src={SOPHISTICATED_PROCESS.stepOne.imageUrl}
            alt="split ribs picture"
            sizes="(max-width: 768px) 277px, 233px
              (min-width: 768px) 634px, 447px
              (min-width: 1200px) 792px, 638px"
          />
        </Box>
        <Box maxW={{ base: '90%', md: '38%' }}>
          <Heading as="h4" size="medium" variant="secondary">
            01.{SOPHISTICATED_PROCESS.stepOne.title}
          </Heading>
          <Text pt={{ base: '8px', md: '28px' }}>
            {SOPHISTICATED_PROCESS.stepOne.description}
          </Text>
        </Box>
      </Flex>
      <Flex
        pt={{ base: '50px', md: '118px' }}
        flexDirection={{ base: 'column', md: 'row-reverse' }}
        justifyContent="space-between"
      >
        <Box>
          <Box
            width={{ base: '277px', md: '455px', '2xl': '650px' }}
            height={{ base: '227px', md: '373px', '2xl': '533px' }}
            position="relative"
            margin="0 auto"
          >
            <Image
              fill
              src={SOPHISTICATED_PROCESS.stepTwo.imageUrl}
              alt="marinated ribs picture"
              sizes="(max-width: 768px) 277px, 227px
                (min-width: 768px) 455px, 373px
                (min-width: 1200px) 650px, 533px"
            />
          </Box>
          <Heading
            as="h4"
            size="medium"
            variant="secondary"
            maxW={{ base: '90%', md: '455px', '2xl': '650px' }}
            pt={{ base: '8px', md: '46px' }}
            pl={{ base: '0px', md: '40px', '2xl': '80px' }}
            margin={{ base: '0 auto', md: 'unset' }}
          >
            02.{SOPHISTICATED_PROCESS.stepTwo.title}
          </Heading>
          <Text
            pt={{ base: '8px', md: '28px' }}
            maxW={{ base: '90%', md: '455px', '2xl': '650px' }}
            pl={{ base: '0px', md: '40px', '2xl': '80px' }}
            margin={{ base: '0 auto', md: 'unset' }}
          >
            {SOPHISTICATED_PROCESS.stepTwo.description}
          </Text>
        </Box>
        <Flex
          pl={{ base: '0px', md: '71px' }}
          pt={{ base: '50px', md: '150px' }}
          flexDirection={{ base: 'column-reverse', md: 'column' }}
          gap={{ base: '10px', md: 'unset' }}
        >
          <Box
            margin={{ base: '0 auto', md: 'unset' }}
            pb={{ base: '20px', md: '46px' }}
            maxW={{ base: '90%', md: '455px', '2xl': '625px' }}
          >
            <Heading as="h4" size="medium" variant="secondary">
              03.{SOPHISTICATED_PROCESS.stepThree.title}
            </Heading>
            <Text pt={{ base: '8px', md: '28px' }}>
              {SOPHISTICATED_PROCESS.stepThree.description}
            </Text>
          </Box>
          <Box
            width={{ base: '277px', md: '455px', '2xl': '650px' }}
            height={{ base: '217px', md: '356px', '2xl': '509px' }}
            position="relative"
            margin="0 auto"
          >
            <Image
              fill
              src={SOPHISTICATED_PROCESS.stepThree.imageUrl}
              alt="grilled ribs picture"
              sizes="(max-width: 768px) 277px, 217px
                (min-width: 768px) 455px, 356px
                (min-width: 1200px) 650px, 509px"
            />
          </Box>
        </Flex>
      </Flex>
    </>
  )
}

export default Process
