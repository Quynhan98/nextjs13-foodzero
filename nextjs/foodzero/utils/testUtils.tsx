import { ReactNode } from 'react'
import { render } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'

// Themes
import { customTheme } from '@themes/index'

// Contexts
import { AuthProvider } from '@contexts/AuthProvider'
import { BookingProvider } from '@contexts/BookingProvider'

export const customRender = (component: ReactNode) => {
  return render(
    <ChakraProvider theme={customTheme}>
      <AuthProvider>
        <BookingProvider>{component}</BookingProvider>
      </AuthProvider>
    </ChakraProvider>,
  )
}

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
