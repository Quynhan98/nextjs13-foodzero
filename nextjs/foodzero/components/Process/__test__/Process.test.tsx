import { render } from '@utils/testUtils'

// Components
import Process from '@components/Process'

describe('Process component', () => {
  it('Renders Process correctly', () => {
    const { container } = render(<Process />)

    expect(container).toMatchSnapshot()
  })
})
