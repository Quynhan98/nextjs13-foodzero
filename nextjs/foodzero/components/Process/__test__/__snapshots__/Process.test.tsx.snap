// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process component Renders Process correctly 1`] = `
<div>
  <div
    class="css-1nwebte"
  >
    <div
      class="css-115fs0x"
    >
      <img
        alt="split ribs picture"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="(max-width: 768px) 277px, 233px (min-width: 768px) 634px, 447px (min-width: 1200px) 792px, 638px"
        src="/_next/image?url=%2Fimages%2FsplitRibs.webp&w=3840&q=75"
        srcset="/_next/image?url=%2Fimages%2FsplitRibs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2FsplitRibs.webp&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <div
      class="css-1xpzrde"
    >
      <h4
        class="chakra-heading css-sqbc2k"
      >
        01.
        Slice
      </h4>
      <p
        class="chakra-text css-8j6lav"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus imperdiet. Turpis egestas ultricies purus auctor tincidunt lacus nunc. Convallis pellentesque quis fringilla sagittis. Egestas in risus sit nunc nunc, arcu donec nam etiam.
      </p>
    </div>
  </div>
  <div
    class="css-gh6j68"
  >
    <div
      class="css-0"
    >
      <div
        class="css-14jff95"
      >
        <img
          alt="marinated ribs picture"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 768px) 277px, 227px (min-width: 768px) 455px, 373px (min-width: 1200px) 650px, 533px"
          src="/_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2FmarinatedRibs.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <h4
        class="chakra-heading css-ysd11w"
      >
        02.
        Pickled
      </h4>
      <p
        class="chakra-text css-lg3a2o"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus imperdiet.
      </p>
    </div>
    <div
      class="css-a5uq8u"
    >
      <div
        class="css-1dnyehg"
      >
        <h4
          class="chakra-heading css-sqbc2k"
        >
          03.
          Bake  
        </h4>
        <p
          class="chakra-text css-8j6lav"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Purus lorem id penatibus imperdiet.
        </p>
      </div>
      <div
        class="css-19jted4"
      >
        <img
          alt="grilled ribs picture"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 768px) 277px, 217px (min-width: 768px) 455px, 356px (min-width: 1200px) 650px, 509px"
          src="/_next/image?url=%2Fimages%2FgrilledRibs.webp&w=3840&q=75"
          srcset="/_next/image?url=%2Fimages%2FgrilledRibs.webp&w=16&q=75 16w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=32&q=75 32w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=48&q=75 48w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=64&q=75 64w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=96&q=75 96w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=128&q=75 128w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=256&q=75 256w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=384&q=75 384w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=640&q=75 640w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=750&q=75 750w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=828&q=75 828w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2FgrilledRibs.webp&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
